### PR TITLE
Update ESPnet-3 from clone Master

### PR DESCRIPTION
**Pull request by Fhrozen**
## Notes

I am using a clone master from my repo to avoid undesired changes in the espnet:master.

## What did you change?

This pull request introduces several improvements and updates to the project's CI/CD workflows, dependency management, and configuration files. The main focus is on modernizing CI pipelines, updating dependencies, introducing a new workflow for Debian 12, and enhancing automation for dependency and pull request management.

**CI/CD Workflow Modernization and Improvements:**

- Added a new GitHub Actions workflow (`ci_on_debian12.yml`) to test ESPnet1 and ESPnet2 on Debian 12, ensuring compatibility with this environment.
- Updated existing CI workflows to use the latest versions of key GitHub Actions, including `actions/cache@v4`, `codecov/codecov-action@v5`, and `peaceiris/actions-gh-pages@v4`, improving reliability and support. [[1]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L40-R45) [[2]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L73-R73) [[3]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L88-R88) [[4]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L102-R107) [[5]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L138-R138) [[6]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L153-R153) [[7]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L164-R164) [[8]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L184-R184) [[9]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L213-R213) [[10]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L227-R232) [[11]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L259-R259) [[12]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L274-R274) [[13]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L285-R285) [[14]](diffhunk://#diff-e9868fff133d797c4c6a9e2ad3cbc3fc55bfd09b1a2cc1324218b5609567475fL18-R22) [[15]](diffhunk://#diff-07e2a7a53f167a4bfe47fd72cc304f5071279a86c297985caaabd11ff35da7b1L24-R24) [[16]](diffhunk://#diff-674587e64048087d979ba6fee9bf577629e68dc73c24a6898547d188cd5a8304L62-R62) [[17]](diffhunk://#diff-44860da54860e14b8008275a29a60e7f13117b31138e1a30b27e7dc466e6b5d0R16-R24)
- Updated the default PyTorch versions tested in CI to include newer releases (2.5.1, 2.6.0, 2.7.1, 2.8.0), and removed older versions, ensuring the project stays current with upstream dependencies. [[1]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L102-R107) [[2]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L227-R232) [[3]](diffhunk://#diff-48d19e7512e7e9ec72c5503d3178348ad59275991f8b8334269329766e343de9L40-R45) [[4]](diffhunk://#diff-e9868fff133d797c4c6a9e2ad3cbc3fc55bfd09b1a2cc1324218b5609567475fL18-R22)
- Fixed a typo in the workflow name (`publich docker image` → `publish docker image`) and set the correct environment for Docker publishing. [[1]](diffhunk://#diff-44860da54860e14b8008275a29a60e7f13117b31138e1a30b27e7dc466e6b5d0L1-R1) [[2]](diffhunk://#diff-44860da54860e14b8008275a29a60e7f13117b31138e1a30b27e7dc466e6b5d0R16-R24)

**Dependency and Automation Enhancements:**

- Introduced a `dependabot.yml` configuration to automate dependency updates for GitHub Actions, scheduled weekly.
- Upgraded pre-commit hooks to use `pre-commit-hooks` v6.0.0 for better linting and code hygiene.

**Automation and Configuration Updates:**

- Added an `automerge.yml` workflow to automate merging pull requests based on labels, review status, and other triggers, streamlining the contribution process.
- Updated `.gemini/config.yaml` to tune code review settings, reducing comment noise and disabling automatic PR summaries.

**Other Notable Updates:**

- Updated `cancel-workflow-action` to v0.12.1 for improved workflow cancellation reliability.
- Minor CI script improvements, such as installing `six` alongside `numba` and removing unused Chainer installer steps.

These changes collectively modernize the project's automation, improve CI reliability, and ensure compatibility with newer environments and dependencies.

### CODE REVIEW

**Comment by sw005320** [comment]
@Masao-Someki, can you review this?

File: espnet/utils/dummy_chainer.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
+"""Dummy classes of chainer required for CI test."""
```
Comment (Masao-Someki) [inline espnet/utils/dummy_chainer.py:1]: I'm not sure if this is a good solution, but if this is required for tests, how about having a one class like the following DummyChainer and use it..?

```
class DummyChainer:
    def __getattr__(self, name):
        def _missing(*args, **kwargs):
            raise NotImplementedError(f"Method '{name}' is not implemented")
        return _missing
```

**Comment by Masao-Someki** [comment]
The CI has issue, but for the espnet-ez/espnet-3 related CI I think we can just ignore it for now. It should be recovered in #6179

**Comment by sw005320** [comment]
LGTM
Once we reflect gemni and @Masao-Someki's reviews, we can merge this PR

File: espnet2/spk/pooling/stat_pooling.py [hunk 2]
```diff
@@ -23,11 +23,40 @@ def __init__(self, input_size: int = 1536):
...
+                ],
+                dim=0,
+            )
+            st = torch.stack(
+                [
+                    torch.std(x[i, :, : int(l.item())], dim=-1, unbiased=False)
+                    for i, l in enumerate(feat_lengths)
+                ],
+                dim=0,
+            )  # unbiased=False (divided by N rather than N - 1)
```
[the remark this answers is not in the document]
Comment (Fhrozen) [reply espnet2/spk/pooling/stat_pooling.py]: @gemini-code-assist Could you provide the code to fit the suggestions in this code section?

File: espnet/utils/dummy_chainer.py (hunk 1 above)
Comment (Fhrozen) [reply espnet/utils/dummy_chainer.py:1]: I think that this could be implemented in a different PR, especially because this PR is only to update espnet3 branch from master. So adding or changing the Dummy chainer will go out of the scope of this PR.

**Comment by Fhrozen** [comment]
@sw005320 @Masao-Someki, I reflected the comments from Gemini, but the comments from Masao should be addressed in a different PR, because the main goal of this PR is only to update the espnet3 to match master updates, and allow espnet3 users to have the updates. I think that any additional modifications should be addressed in a different PR. And as Masao mentioned, the CI issues are already being covered in a different PR related to espnet3.

**Comment by sw005320** [comment]
@Qingzheng-Wang, can you review the pooling part?
Gemini suggests an efficient implementation, but I want to ensure that it is correct.

**Comment by sw005320** [comment]
This should be an issue
https://github.com/espnet/espnet/actions/runs/17598784475/job/49996688545?pr=6231#step:7:4033

**Comment by Fhrozen** [comment]
@sw005320 This issues is expected because the current ESPnet3 uses numpy=1.23 in the pyproject.toml https://github.com/Fhrozen/espnet/blob/736ff42b6ce6f80261ac930959732765bfc6826e/pyproject.toml#L33
I think that it should be updated by Masao in a different PR.

File: espnet2/spk/pooling/chn_attn_stat_pooling.py [hunk 3]
```diff
@@ -24,31 +27,72 @@ def __init__(self, input_size: int = 1536):
...
+            masked_x = x.masked_fill(~mask, 0)
+            # sum over time
+            sum_val = masked_x.sum(dim=-1, keepdim=True)
+            sum_sq_val = (masked_x**2).sum(dim=-1, keepdim=True)
+            # feat_lengths might be 0, add epsilon
+            feat_lengths_ = feat_lengths.view(-1, 1, 1).clamp(min=1e-8)
+            mean = sum_val / feat_lengths_
+            # var = E[X^2] - (E[X])^2
+            var = sum_sq_val / feat_lengths_ - mean**2
+            std = torch.sqrt(var.clamp(min=1e-4, max=1e4))
```
Comment (Qingzheng-Wang) [inline espnet2/spk/pooling/chn_attn_stat_pooling.py]: It is not necessary to clamp the max value. Clamping the minimum value is sufficient to ensure numerical stability and avoid negative variances.
```suggestion
            std = torch.sqrt(var.clamp(min=1e-4))

File: espnet2/spk/pooling/stat_pooling.py [hunk 4]
```diff
@@ -23,11 +23,41 @@ def __init__(self, input_size: int = 1536):
...
+
+            # Calculate mean over the time dimension (dim=-1)
+            mu = (x * mask).sum(dim=-1) / feat_lengths.unsqueeze(1)
+
+            # Calculate standard deviation over the time dimension (dim=-1)
+            # unbiased=False (divided by N rather than N - 1)
+            variance = ((x - mu.unsqueeze(-1)) ** 2 * mask).sum(
+                dim=-1
+            ) / feat_lengths.unsqueeze(1)
+            st = torch.sqrt(variance.clamp(min=1e-4, max=1e4))
```
Comment (Qingzheng-Wang) [inline espnet2/spk/pooling/stat_pooling.py]: It is not necessary to clamp the max value. Clamping the minimum value is sufficient to ensure numerical stability and avoid negative variances.
```suggestion
            st = torch.sqrt(variance.clamp(min=1e-4))

**Comment by sw005320** [comment]
@Masao-Someki, please fix the CI errors
